### PR TITLE
Transaction simulation

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes BX-####
Figma link (if any): https://www.figma.com/file/a33TH3fYEWNC6xTePxZ8sN/BX-Signing?type=design&node-id=340%3A2063&mode=dev

## Screen recordings / screenshots

https://github.com/rainbow-me/browser-extension/assets/6232729/784ce14e-af5f-4a5e-bac4-a097183adc74

https://github.com/rainbow-me/browser-extension/assets/6232729/c4203e89-4292-40ee-ba6b-0377b985ba01

https://github.com/rainbow-me/browser-extension/assets/6232729/192e7157-d83e-4a38-8887-2bc469354e07

### CODE REVIEW

File: src/entries/popup/pages/messages/SendTransaction/SendTransactionsInfo.tsx [hunk 1]
```diff
@@ -1,153 +1,341 @@
+/* eslint-disable @typescript-eslint/ban-types */
```
Comment (estebanmino) [inline src/entries/popup/pages/messages/SendTransaction/SendTransactionsInfo.tsx:1]: why did you need this?

File: src/entries/popup/pages/messages/SignMessage/index.tsx [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
+import { Navigate } from 'react-router-dom';
 
 import { analytics } from '~/analytics';
 import { event } from '~/analytics/event';
 import { i18n } from '~/core/languages';
 import { useDappMetadata } from '~/core/resources/metadata/dapp';
 import { useFeatureFlagsStore } from '~/core/state/currentSettings/featureFlags';
 import { ProviderRequestPayload } from '~/core/transports/providerRequestTransport';
-import { ChainId } from '~/core/types/chains';
+// import { ChainId } from '~/core/types/chains';
```
Comment (estebanmino) [inline src/entries/popup/pages/messages/SignMessage/index.tsx:10]: 👀

File: src/entries/popup/pages/messages/SignMessage/index.tsx [hunk 3]
```diff
@@ -55,7 +57,7 @@ export function SignMessage({
   const { activeSession } = useAppSession({ host: dappMetadata?.appHost });
   const { watchedWallets } = useWallets();
 
-  const selectedChainId = activeSession?.chainId ?? ChainId.mainnet;
+  // const selectedChainId = activeSession?.chainId ?? ChainId.mainnet;
```
Comment (estebanmino) [inline src/entries/popup/pages/messages/SignMessage/index.tsx:60]: 👁️

File: src/entries/popup/pages/messages/SignMessage/index.tsx [hunk 4]
```diff
@@ -145,18 +147,31 @@ export function SignMessage({
     }
   }, [featureFlags.full_watching_wallets, isWatchingWallet, rejectRequest]);
 
+  if (!selectedWallet || !dappMetadata) return <Navigate to="/" />;
```
Comment (estebanmino) [inline src/entries/popup/pages/messages/SignMessage/index.tsx:150]: not sure how i feel about this, these values can be undefined? 

if so we would be sending the user to home instead of rejecting the message?

File: src/entries/popup/pages/messages/Simulation.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
+/* eslint-disable @typescript-eslint/ban-types */
```
Comment (estebanmino) [inline src/entries/popup/pages/messages/Simulation.tsx:1]: why did you need this?

File: src/entries/popup/pages/messages/useSimulateTransaction.tsx [hunk 6]
```diff
@@ -0,0 +1,164 @@
+/* eslint-disable @typescript-eslint/ban-types */
```
Comment (estebanmino) [inline src/entries/popup/pages/messages/useSimulateTransaction.tsx:1]: why did you need this?

**Comment by estebanmino** [comment]
@greg-schrammel could you link figma? 🙏

**Review by estebanmino — COMMENTED** [review]
https://github.com/rainbow-me/browser-extension/assets/12115171/8548723d-dbce-40cf-a892-001dafaea2be

i feel like there's something odd while switching tabs?

it moves left and right very fast twice while switching

**Comment by estebanmino** [comment]
this is 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥

File: src/entries/popup/pages/messages/SignMessage/index.tsx (hunk 4 above)
Comment (greg-schrammel) [reply src/entries/popup/pages/messages/SignMessage/index.tsx]: removed it, something super weird must happen to get into a state where the user is in this screen with no active dapp session, so I'm assuming this wont happen :)

**Review by estebanmino — COMMENTED** [review]
image

image


found this with test dapp, the message is not entirely visible and details is empty, not sure if we want to hide that tab when this happens

**Review by estebanmino — APPROVED** [review]

**Review by derHowie — APPROVED** [review]
🔥
